pkg/provision/kubernetes.go

```diff
@@ -0,0 +1,311 @@
+package provision
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/threefoldtech/zos/pkg"
+	"github.com/threefoldtech/zos/pkg/stubs"
+)
+
+// Kubernetes reservation data
+type Kubernetes struct {
+	// Size of the vm, this defines the amount of vCpu, memory, and the disk size
+	Size uint8 `json:"size"`
+
+	// NetworkID of the network namepsace in which to run the VM. The network
+	// must be provisioned previously.
+	NetworkID pkg.NetID `json:"network_id"`
+	// IP of the VM. The IP must be part of the subnet available in the network
+	// resource defined by the networkID on this node
+	IP net.IP `json:"ip"`
+
+	// ClusterSecret is the hex encoded encrypted cluster secret.
+	ClusterSecret string `json:"cluster_secret"`
+	// MasterIPs define the URL's for the kubernetes master nodes. If this
+	// list is empty, this node is considered to be a master node.
+	MasterIPs []net.IP `json:"master_ips"`
+	// SSHKeys is a list of ssh keys to add to the VM. Keys can be either
+	// a full ssh key, or in the form of `github:${username}`. In case of
+	// the later, the VM will retrieve the github keys for this username
+	// when it boots.
+	SSHKeys []string `json:"ssh_keys"`
+
+	PlainClusterSecret string `json:"-"`
+}
+
+// TODO: move flist
+const k3osFlistURL = "https://hub.grid.tf/lee/k3os.flist"
+
+func kubernetesProvision(ctx context.Context, reservation *Reservation) (interface{}, error) {
+	return nil, kubernetesProvisionImpl(ctx, reservation)
```

Comment:
The result of the deployment doesn't return any information ?
How does the user can reach its cluster with kubectl once it is deploy then ?

Reply:
I assume this part was missed out in implementation. What information is needed to be returned to the User so I can add it?

Reply:
Any information the sysadmin requires to be able to use kubectl against the cluster. I do not know what this is.

Reply:
it is the file generated by k3s when launched for the first time it is  written to /etc/rancher/k3s/k3s.yaml

Reply:
Hmm, that is gonna be tricky to read a file from inside the VM. only idea that comes to mind is to mount the raw disk in in `ro` mode and then try to read that file.

Reply:
We decided that the user can use his ssh key to retrieve the required file